src/WebApp/Components/Chatbot/Chatbot.razor

```diff
@@ -8,8 +8,8 @@
 @inject CatalogService CatalogService
 @inject BasketState BasketState
 @inject AuthenticationStateProvider AuthenticationStateProvider
-@inject IConfiguration Configuration
 @inject ILoggerFactory LoggerFactory
+@inject IServiceProvider ServiceProvider
```

Comment:
Does razor have any notion of an `@inject` that's optional, i.e. the equivalent of GetService rather than GetRequiredService? If yes, then this could be that with Kernel, and then the code below doesn't need to call GetService and can just do what it does subsequently, checking it for null.

Reply:
Nevermind, I see your comment/question above now: https://github.com/dotnet/eShop/pull/147#discussion_r1442177118.